integration_tests/run.sh

```diff
@@ -28,61 +28,221 @@
+        # Prints output to stdout and to an output file.
+        behave --no-logcapture --no-color --no-capture -D datapath=/data/test_data -i "${prefix}2grid.feature"\
+         --format pretty --format json.pretty 2>&1 | tee "$test_details"
+
+        format_test_details "$test_details"
+
+        # Replace FAILED with SUCCESSFUL.
+        save_vars "${prefix:0:1}2g_tests=SUCCESSFUL"
+    )
+}
+
+create_documentation()
+{
+    prefix=$1
+    package_name=$2
     # Make docs.
     cd "$WORKSPACE"/doc
     make latexpdf POLAR2GRID_DOC="$prefix"
-    cp "$WORKSPACE"/doc/build/latex/*.pdf "$package_name"
+    # Copy pdfs to package directory.
+    cp "$WORKSPACE"/doc/build/latex/*.pdf "${WORKSPACE}/$package_name"
     # Clear out intermediate results and rebuild for HTML document.
     make clean
     # Needs to be second since Jenkins makes an html in workspace from the file generated by this command.
     make html POLAR2GRID_DOC="$prefix"
-    cp -r "$WORKSPACE"/doc/build/html "$package_name"
-    chmod -R a+rX "$package_name"
-done
+    # Copy html to package directory.
+    cp -r "$WORKSPACE"/doc/build/html "${WORKSPACE}/$package_name"
+    # Replaces FAILED with SUCCESSFUL.
+    save_vars "${prefix:0:1}2g_documentation=SUCCESSFUL"
+}
+
+# Copies ("publishes") tarball and documentation to bumi:/tmp and give the ability for others to copy it.
+publish_package()
+{
+    prefix=$1
+    package_name=$2
+    # Remove the directory if it was already made.
+    rm -rf "/tmp/$package_name"
+    cp -r "${WORKSPACE}/$package_name" "/tmp/$package_name"
+    chmod -R a+rX "/tmp/$package_name"
+    save_vars "${prefix:0:1}2g_package_published=TRUE"
+}
+
+set -x
+
+# Allows the program to set finish_time while also returning a failing code.
+exit_status=0
+
+try
+(
+    set -e
+    setup_vars
+    setup_conda
+
+    # Make polar2grid and geo2grid separately.
+    for prefix in ${prefixes}; do
+        try
+        (
+            set -e
+            swbundle_name="${WORKSPACE}/${prefix}2grid-swbundle-${suffix}"
+            cd "$WORKSPACE"
+            conda activate jenkins_p2g_swbundle
+            "$WORKSPACE"/create_conda_software_bundle.sh "$swbundle_name"
+
+            package_name="${prefix}2grid-${suffix}"
+            mkdir "${WORKSPACE}/$package_name"
+            # Copies tarball to package directory.
+            cp "${swbundle_name}.tar.gz" "${WORKSPACE}/$package_name"
+
+            conda activate jenkins_p2g_docs
+            if [[ "$commit_message" =~ (^|.[[:space:]])"["([pg]2g-)?skip-tests"]"$ ]]; then
+                # Replace FAILED with SKIPPED.
+                save_vars "${prefix:0:1}2g_tests=SKIPPED"
+            else
+                run_tests $prefix
+                update_exit_status ${exit_status}
+            fi
+            create_documentation $prefix $package_name
+            # Only publishes if both tests and documentation passed.
+            if [[ ${exit_status} -eq 0 ]]; then
```

Comment:
Is this actually checking the documentation exit status at all?

Reply:
It is now. If documentation fails, then the code will exit the subshell before trying to publish.